cli/provider_cache.go

```diff
@@ -255,22 +254,15 @@ func (cache *ProviderCache) createLocalCLIConfig(ctx context.Context, opts *opti
 	for _, registryName := range opts.ProviderCacheRegistryNames {
 		providerInstallationIncludes = append(providerInstallationIncludes, registryName+"/*/*")
 
-		urls, err := DiscoveryURL(ctx, registryName)
+		apiURLs, err := cache.Server.DiscoveryURL(ctx, registryName)
 		if err != nil {
-			if !liberrors.As(err, &NotFoundWellKnownURL{}) {
-				return err
-			}
-
-			urls = DefaultRegistryURLs
-			opts.Logger.Debugf("Unable to discover %q registry URLs, reason: %q, use default URLs: %s", registryName, err, urls)
-		} else {
-			opts.Logger.Debugf("Discovered %q registry URLs: %s", registryName, urls)
+			return err
```

Comment:
Make sense to wrap `errors.WithStackTrace(err)` ?

Reply:
> Does it make sense to wrap `errors.WithStackTrace(err)`?

No, because `apiURLs, err := cache.Server.DiscoveryURL(ctx, registryName)` already wrapped the error, so we'll have the full stack trace anyway. Wrapping the error only makes sense when we create an error or grab it from a third-party package.